### PR TITLE
Updates on VF2++

### PR DESCRIPTION
**Pull request by kpetridis24**
This is the second post from my GSoC 2022 journey. It includes updates of my progress so far. More updates in the next posts.

### CODE REVIEW

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (tupui) [inline content/posts/networkx/node-ordering-Ti-updating/index.md:14]: Add a link to the previous post. You can have relative links `[here]({{< relref "../GSoC-2022" >}}`.

I would also update the first post with a table of content at the beginning. We can have the same table in all posts. Something similar is found in other blogs when there are follow up.

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (tupui) [inline content/posts/networkx/node-ordering-Ti-updating/index.md:72]: Is there a reason you did not use normal markdown here?
```suggestion
![Illustration of $T_i$.](Ti.png)
```

**Review by tupui — COMMENTED** [review]
Thank you for the update @kpetridis24.

As @rossbar suggested previously, we should probably group all these posts under a `gsoc` folder (either in `posts/` or `posts/networkx`).

Otherwise it reads well. Not sure how much detail we want to give. I left some suggestions.

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (kpetridis24) [reply content/posts/networkx/node-ordering-Ti-updating/index.md:72]: nope, i just took it as is from another post. I am not an md expert unfortunately :)

**Comment by kpetridis24** [comment]
> Thank you for the update @kpetridis24.
> 
> As @rossbar suggested previously, we should probably group all these posts under a `gsoc` folder (either in `posts/` or `posts/networkx`).
> 
> Otherwise it reads well. Not sure how much detail we want to give. I left some suggestions.

I could provide a lot more detail but i didn't consider it necessary. I took a look at previous posts and they also kind of include details on the project and code from the implementations. So i thought that some detailed information would probably fit.

**Comment by kpetridis24** [comment]
@tupui @rossbar I added a diagram showing the performance improvement when using the incremental updating and i also included some complexity analysis. I tried to keep the complexity section VF2++-agnostic, particularly i just presented the two functions and computed the bigO for both of them, something the reader can do as well without any previous knowledge on the implementation. Feel free to give any feedback.

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (tupui) [inline content/posts/networkx/node-ordering-Ti-updating/index.md:155]: The conclusion should stand out
```suggestion
where $M_{T_1}$ is the expected (average) number of elements in $T_1$.

Certainly, the complexity is much better in this
case, as $D$ and $M_{T_1}$ are significantly smaller than $N_mD$ and $N$.
```

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (tupui) [reply content/posts/networkx/node-ordering-Ti-updating/index.md:91]: Thank you for the update. If you can regenerate the figure, we need to make it more accessible by using different linestyles and adding a legend.

**Comment by kpetridis24** [comment]
@tupui Added different linestyles and legend!

File: Makefile
Comment (tupui) [inline Makefile:39]: Revert this too.
```suggestion
	@hugo --printI18nWarnings server
```

**Review by tupui — COMMENTED** [review]
LGTM, I will let @rossbar or another NetworkX maintainer have a look before merging.

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (rossbar) [inline content/posts/networkx/node-ordering-Ti-updating/index.md:99]: I think this statement doesn't *quite* match what the performance plot shows. For instance, both methods *appear* to have an exponential increase in run-time vs. number of nodes, it's just that the incremental update method increases much less rapidly. Perhaps something like:

```suggestion
$0.7$. It can clearly be seen that execution time of the brute force
method increases much more rapidly with the number of nodes/edges than
the incremental update method, as expected.
```

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (rossbar) [inline content/posts/networkx/node-ordering-Ti-updating/index.md:157]: Since you're already working on the next post, it might be nice to wrap this one up in a sentence or two and give readers a teaser about what's coming next! e.g. `In this post we've investigated how node ordering is implemented for the VF2++ algorithm; next we'll investigate another important compenent: cuts and consistency checks <or whatever your next post will be about)`

**Review by rossbar — COMMENTED** [review]
This is great, thanks @kpetridis24 !

I've left just two final suggestions (and opened a PR against your fork with some minor grammatical/formatting changes). Other than that I think this is good to go - great job summarizing your work on this component of the VF2++ algorithm!

File: content/posts/networkx/node-ordering-Ti-updating/index.md
Comment (kpetridis24) [reply content/posts/networkx/node-ordering-Ti-updating/index.md:99]: Sure, you are right!

**Review by rossbar — APPROVED** [review]
> The formatting destroyed the pseudocodes :')

Yeah, I don't know what happened in 1f86fd3... whatever it was it was a nasty change.

Anyways, I went ahead and reverted that mess, cherry-picked out the conclusion that you added and re-committed it (w/ a grammatical edit or two) in 5486851. So the formatting errors should be fixed and all of your content present and accounted for!

Thanks @kpetridis24 for writing it up!

**Pull request merged**
